File: C++/383. Ransom Note.cpp
#include <iostream>
#include <vector>
#include<unordered_map>

using namespace std;


class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> HashCounterOfMagazine;

        for(char c: magazine){
            HashCounterOfMagazine[c]++;
        }

        for (char c: ransomNote){
            if(HashCounterOfMagazine[c] <= 0){
                return false;
            }
            HashCounterOfMagazine[c]--;
        }
        
        return true;
    }    
};

File: Python/383. Ransom Note.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        DictOfLetters = {}

        for letter in magazine:
            DictOfLetters[letter] = DictOfLetters.get(letter, 0) + 1
        for letter in ransomNote:
            if DictOfLetters.get(letter) <= 0:
                return False
            DictOfLetters[letter] -= 1
        
        return True
